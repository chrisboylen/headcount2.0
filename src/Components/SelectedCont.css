.compare-cont {
  background-color: #3c5354;
  border-radius: 20px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  display: flex;
  padding: 2em 1em 0 1em;
  margin: 2em;
}

.compare-info {
  align-items: center;
  background-color: #f7b488;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  color: #2a393a;
  display: flex;
  flex: 1 18em;
  flex-direction: column;
  justify-content: space-around;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}
 
span {
  display: block;
}