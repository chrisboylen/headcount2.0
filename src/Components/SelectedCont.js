import React from 'react';
import Card from './Card';
import './SelectedCont.css';
import PropTypes from 'prop-types';

const SelectedCont = ({ selectedCards, selectCard, compareDistrictAverages }) => {
  const displaySelectedCards = Object.values(selectedCards).map((district, index) => {
    return  <Card location={district.location}
      stats={district.stats}
      isSelected={district.isSelected}
      key={`sel-card${index}`}
      selectCard={selectCard}
    />;
  });

  const compareSelectedCards = () => {
    if (selectedCards.length === 2) {
      const location1 = selectedCards[0].location;
      const location2 = selectedCards[1].location;
      const comparedData = compareDistrictAverages(location1, location2); 
    
      return (
        <div className="compare-cont">
          {displaySelectedCards[0]}
          <div className="compare-info">
            <h2>{location1} : {comparedData[location1]}</h2>
            <h2><span>Compared Average</span>{comparedData.compared}</h2>
            <h2>{location2} : {comparedData[location2]}</h2>
          </div>
          {displaySelectedCards[1]}
        </div>
      );
    } else {
      return displaySelectedCards;
    }
  };

  return (
    <article className="card-cont">
      {compareSelectedCards()}
    </article>
  );
};

SelectedCont.propTypes = {
  selectedCards: PropTypes.array,
  data: PropTypes.array,
  selectCard: PropTypes.func, 
  compareDistrictAverages: PropTypes.func
};

export default SelectedCont;