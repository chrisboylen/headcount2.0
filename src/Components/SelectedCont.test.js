import React from 'react';
import { shallow, mount } from 'enzyme';
import SelectedCont from './SelectedCont';

describe('SELECTEDCONT', () => {
  it('Should match snapshot when there is an empty array', () => {
    const wrapper = shallow (
      <SelectedCont
        selectedCards={[]}
      />
    )

    expect(wrapper.html()).toMatchSnapshot();
  })
  
  it('Should match snapshot when there is data in the array', () => {
    const mockData = [
      {
        location: 'COLORADO',
        isSelected: true,
        stats: {
          2002: 1
        }, 
      },
      {
        location: 'BINGO',
        isSelected: true,
        stats: {
          2002: 1
        }
      }
    ];
    
    const compareDistrictAveragesMock = jest.fn()

    const wrapper = shallow(
      <SelectedCont 
        selectedCards={mockData} 
        compareDistrictAverages={compareDistrictAveragesMock}
      />
    );

    expect(wrapper.html()).toMatchSnapshot();
  })
})