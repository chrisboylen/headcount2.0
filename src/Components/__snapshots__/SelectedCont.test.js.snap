// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SELECTEDCONT Should match snapshot when there is an empty array 1`] = `"<article class=\\"card-cont\\"></article>"`;
